src/Widgets/VPN/VPNMenuItem.vala

```diff
@@ -91,11 +89,7 @@ public class Network.VPNMenuItem : Gtk.ListBoxRow {
         update ();
 
         connect_button.clicked.connect (() => {
-            if (state == State.CONNECTED_VPN) {
-                deactivate_vpn ();
-            } else {
-                activate_vpn ();
-            }
+            activate ();
```

Comment:
You can use a shorter syntax here
`connect_button.clicked.connect (() => activate ());`